### PR TITLE
528 remove deprecated prior parsing

### LINKED ISSUE #528: Remove `_deprecated_prior_parsing` from `injection_parameters.py`
Self-explanatory.

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Closes #528 

- [x] Clean up `injection_parameters.py`
- [x] Check tests
- [x] Check examples

### CODE REVIEW

**Comment by Rodrigo-Tenorio** [comment]
Rebasing on top of #533 to fix test suite.

**Comment by Rodrigo-Tenorio** [comment]
I think it makes sense to fix codecov in this case:
- [ ] Add test to ensure old API raises exception.

File: pyfstat/injection_parameters.py [hunk 1]
```diff
@@ -122,12 +122,12 @@ class InjectionParametersGenerator:
...
-          | (``np.random.Generator`` type)
-          | to generate any required random number within the function.
-          | The ``generator`` kwarg is *required* regardless of whether this is a
-          | deterministic or random function.
-          | For example, a negative log-distributed random number could be constructed as
+           | a ``generator`` kwarg as one of its arguments and use such a generator
+           | (``np.random.Generator`` type)
+           | to generate any required random number within the function.
+           | The ``generator`` kwarg is *required* regardless of whether this is a
+           | deterministic or random function.
```
Comment (dbkeitel) [inline pyfstat/injection_parameters.py:129]: old help text, but I found it confusing on re-read. Maybe clearer:
```suggestion
           | to generate any required random number within the function.
           | (A deterministic function could then simply ignore this kwarg.)
```

**Comment by dbkeitel** [comment]
Codecov is running into Simpson's paradox here. :laughing:

Merging.

**Review by dbkeitel — APPROVED** [review]

**Pull request merged**